static/app/utils/replays/hooks/useReplayData.tsx

```diff
@@ -96,143 +81,119 @@ function useReplayData({replaySlug, orgSlug}: Options): Result {
+  const [replayRecord, setReplayRecord] = useState<ReplayRecord>();
 
-  // Fetch every field of the replay. We're overfetching, not every field is needed
+  // Fetch every field of the replay. We're overfetching, not every field is used
   const fetchReplay = useCallback(async () => {
     const response = await api.requestPromise(
       `/projects/${orgSlug}/${projectSlug}/replays/${replayId}/`
     );
-    return response.data;
+    const mappedRecord = mapResponseToReplayRecord(response.data);
+    setReplayRecord(mappedRecord);
   }, [api, orgSlug, projectSlug, replayId]);
 
-  const fetchAttachments = useCallback(
-    async (replayRecord: ReplayRecord) => {
-      if (!replayRecord.count_segments) {
-        return;
-      }
-
-      const baseUrl = `/projects/${orgSlug}/${projectSlug}/replays/${replayId}/recording-segments/`;
-      const perPage = 100;
-
-      const pages = Math.ceil(replayRecord.count_segments / 100);
-      const cursors = new Array(pages)
-        .fill(0)
-        .map((_, i) => `${perPage}:${i}:${i === 0 ? 1 : 0}`);
-
-      await Promise.allSettled(
-        cursors.map(cursor => {
-          const promise = api.requestPromise(
-            `${baseUrl}?download&per_page=${perPage}&cursor=${cursor}`
-          );
-          promise.then(response => {
-            const attachments = responsesToAttachments([response]);
-            setState(prev => ({
-              ...prev,
-              attachments: (prev.attachments ?? []).concat(attachments),
-              errors: prev.errors ?? [],
-            }));
-          });
-          return promise;
-        })
-      );
-    },
-    [api, orgSlug, projectSlug, replayId]
-  );
-
-  const fetchErrors = useCallback(
-    async (replayRecord: ReplayRecord) => {
-      if (!replayRecord.error_ids.length) {
-        return;
-      }
-
-      // Clone the `finished_at` time and bump it up one second because finishedAt
-      // has the `ms` portion truncated, while replays-events-meta operates on
-      // timestamps with `ms` attached. So finishedAt could be at time `12:00:00.000Z`
-      // while the event is saved with `12:00:00.450Z`.
-      const finishedAtClone = new Date(replayRecord.finished_at);
-      finishedAtClone.setSeconds(finishedAtClone.getSeconds() + 1);
-
-      const chunks = chunk(replayRecord.error_ids, ERRORS_PER_PAGE);
-      await Promise.allSettled(
-        chunks.map(errorIds => {
-          const promise = api.requestPromise(
-            `/organizations/${orgSlug}/replays-events-meta/`,
-            {
-              query: {
-                start: replayRecord.started_at.toISOString(),
-                end: finishedAtClone.toISOString(),
-                query: `id:[${String(errorIds)}]`,
-              },
-            }
-          );
-          promise.then(response => {
-            setState(prev => {
-              if (!prev.attachments?.length) {
-                return prev;
-              }
```

Comment:
This was a dumb check i think. It meant that we had to start with attachment data, which shouldn't matter when we're fetching error info. 

It looks like it was a hold-over from earlier when fetching was more sequential.